storage/fanout.go

```diff
@@ -309,11 +318,27 @@ func (q *mergeGenericQuerier) LabelValues(name string) ([]string, Warnings, erro
 	return mergeStringSlices(results), warnings, nil
 }
 
+func (q *mergeGenericQuerier) MarkAsFailedSet(set genericSeriesSet) {
+	q.failedQueriers[q.setQuerierMap[set]] = struct{}{}
+}
+
+func (q *mergeGenericQuerier) DoesBelongToPrimary(set genericSeriesSet) bool {
```

Comment:
does those have to be exported? :thinking:

Reply:
and if yes, let's comment

Reply:
Same question from me, these all appear to be internal stuff

Reply:
It doesn't have to be. I've just followed the same fashion with `IsFailedSet`. It's also not used outside `fanout`. Maybe be better to un-export `IsFailedSet`?

Reply:
I've removed and inlined `IsFailedSet` as well.